### PR TITLE
 Close techspecs with expired results #1261

### PR DESCRIPTION
**Pull request by maslenitsa93**
#1261

### CODE REVIEW

File: libraries/chain/database_worker_objects.cpp [hunk 1]
```diff
@@ -209,18 +209,32 @@ namespace golos { namespace chain {
...
+                continue;
+            }
+
             const auto& wto_post = get_comment(itr->post);
             if (wto_post.created + mprops.worker_techspec_approve_term_sec > now) {
                 break;
             }
 
-            if (itr->state != worker_techspec_state::created) {
+            close_worker_techspec(*itr, worker_techspec_state::closed_by_witnesses);
```
Comment (afalaleev) [inline libraries/chain/database_worker_objects.cpp]: Techspec is not closed by witnesses.
It is closed by timeout.

File: libraries/chain/database_worker_objects.cpp [hunk 2]
```diff
@@ -209,18 +209,32 @@ namespace golos { namespace chain {
...
                 continue;
             }
 
-            close_worker_techspec(*itr, worker_techspec_state::closed_by_witnesses);
+            const auto& result_post = get_comment(result_itr->worker_result_post);
+            if (result_post.created + mprops.worker_result_approve_term_sec > now) {
+                break;
+            }
+
+            close_worker_techspec(*result_itr, worker_techspec_state::closed_by_witnesses);
```
Comment (afalaleev) [inline libraries/chain/database_worker_objects.cpp]: The same as previous

File: libraries/chain/worker_evaluators.cpp [hunk 3]
```diff
@@ -327,8 +327,8 @@ namespace golos { namespace chain {
 
         const auto& wpo = _db.get_worker_proposal(wto.worker_proposal_post);
 
-        GOLOS_CHECK_LOGIC(wto.state == worker_techspec_state::complete || wto.state == worker_techspec_state::payment
-                || wto.state == worker_techspec_state::work,
+        GOLOS_CHECK_LOGIC(wto.state == worker_techspec_state::wip || wto.state == worker_techspec_state::work
+                || wto.state == worker_techspec_state::complete || wto.state == worker_techspec_state::payment,
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: Why it is in this PR?

Comment (maslenitsa93) [reply libraries/chain/worker_evaluators.cpp]: @afalaleev It fixes missed by merges.

File: libraries/chain/database_worker_objects.cpp (hunk 1 above)
Comment (maslenitsa93) [reply libraries/chain/database_worker_objects.cpp]: @afalaleev Do we need special state - closed_by_timeout?

File: libraries/chain/worker_evaluators.cpp (hunk 3 above)
Comment (afalaleev) [reply libraries/chain/worker_evaluators.cpp]: Ok. Create additional issue for this.

**Pull request merged**